lib/Service/TemplateFieldService.php

```diff
@@ -14,16 +14,22 @@
 use OCP\Files\Template\Field;
 use OCP\Files\Template\FieldType;
 use OCP\Http\Client\IClientService;
+use OCP\ICache;
+use OCP\ICacheFactory;
 use Psr\Log\LoggerInterface;
 
 class TemplateFieldService {
+	private ICache $cache;
+
 	public function __construct(
 		private IClientService $clientService,
 		private CapabilitiesService $capabilitiesService,
 		private AppConfig $appConfig,
 		private IRootFolder $rootFolder,
-		private LoggerInterface $logger
+		private LoggerInterface $logger,
+		private ICacheFactory $cacheFactory
 	) {
+		$this->cache = $cacheFactory->createLocal('richdocuments_templates/');
```

Comment:
For this case it does not matter as the createLocal method is rather lightweight, but generally it is a good pattern to avoid calling any code in constructors for classes that use dependency injection. That way this code will always be called as soon as TemplateFieldService initialized which can potentially cause performance issues that are rather hard to debug.

Reply:
Thanks for the note, is there a better pattern I should follow then in the future?